web/packages/extension/src/background.ts

```diff
@@ -9,6 +9,75 @@ async function contentScriptRegistered() {
 }
 
 async function enable() {
+    if (utils.declarativeNetRequest) {
+        const playerPage = utils.runtime.getURL("/player.html");
+        const rules = [
+            {
+                id: 1,
+                action: {
+                    type: chrome.declarativeNetRequest.RuleActionType.REDIRECT,
+                    redirect: { regexSubstitution: playerPage + "#\\0" },
+                },
+                condition: {
+                    regexFilter: ".*",
+                    responseHeaders: [
+                        {
+                            header: "content-type",
+                            values: [
+                                "application/x-shockwave-flash",
+                                "application/futuresplash",
+                                "application/x-shockwave-flash2-preview",
+                                "application/vnd.adobe.flash.movie",
+                            ],
+                        },
+                    ],
+                    resourceTypes: [
+                        chrome.declarativeNetRequest.ResourceType.MAIN_FRAME,
```

Comment:
In `chrome`, this is part of an enum: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chrome/index.d.ts#L12180. I can't use `utils.declarativeNetRequest.ResourceType.MAIN_FRAME` because in `browser` it's a `type` instead: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webextension-polyfill/namespaces/declarativeNetRequest.d.ts#L21